x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/ip/IpPrefixTests.java

```diff
@@ -33,37 +31,66 @@ public IpPrefixTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testC
     public static Iterable<Object[]> parameters() {
 
         var suppliers = List.of(
+            // V4
             new TestCaseSupplier(
                 List.of(DataTypes.IP, DataTypes.INTEGER),
                 () -> new TestCaseSupplier.TestCase(
                     List.of(
                         new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("1.2.3.4"), DataTypes.IP, "ip"),
-                        new TestCaseSupplier.TypedData(24, DataTypes.INTEGER, "prefixLength")
+                        new TestCaseSupplier.TypedData(24, DataTypes.INTEGER, "prefixLengthV4"),
+                        new TestCaseSupplier.TypedData(127, DataTypes.INTEGER, "prefixLengthV6")
                     ),
-                    "IpPrefixEvaluator[ip=Attribute[channel=0], prefixLength=Attribute[channel=1]]",
+                    "IpPrefixEvaluator[ip=Attribute[channel=0], prefixLengthV4=Attribute[channel=1], prefixLengthV6=Attribute[channel=2]]",
                     DataTypes.IP,
                     equalTo(EsqlDataTypeConverter.stringToIP("1.2.3.0"))
                 )
             ),
+            new TestCaseSupplier(
+                List.of(DataTypes.IP, DataTypes.INTEGER),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(EsqlDataTypeConverter.stringToIP("1.2.3.4"), DataTypes.IP, "ip"),
+                        new TestCaseSupplier.TypedData(24, DataTypes.INTEGER, "prefixLengthV4")
+                    ),
+                    "IpPrefixOnlyV4Evaluator[ip=Attribute[channel=0], prefixLengthV4=Attribute[channel=1]]",
+                    DataTypes.IP,
+                    equalTo(EsqlDataTypeConverter.stringToIP("1.2.3.0"))
```

Comment:
nit: I'd still be more confident in the tests if we used a randomized ipv4, and would compare against a simple reference implementation with a random prefix length. The reference impl. would be easier to validate as it'd be free from evaluator-specific code, optimizations, checking against the ipv4 prefix...

Also applies below.